### DISCUSSION TITLE
Errors

### DISCUSSION THREAD

**Discussion by XmeGit** [lucky-media/landtamic#5] [General]
Hello! After the installation I tested Landtamic. 
I came about the following errors: 
- Globals: Theme didn't work at all; Error "The given data was invalid" no matter what I was doing.
- Pages: Preview didn't work when I made several changes at once. Could not exactly locate the problems. 
"Contact" throws: Facade\Ignition\Exceptions\ViewException
A form handle is required on Form tags. Please refer to the docs for more information. 
Gone after deactivating. 

Is the theme production ready? What I shall I do? 
I am new to Statamic and the Community?
Thanks!

**Comment by lokmanm** [comment]
Hi @XmeGit,

In order to help you I need to know exactly the errors that you were encountering. 

1. Globals: That error is usually when you give wrong data to the wrong field, it should indicate it with `red`.  If you let me know which field it is I can let you know the error.

2. Pages: Preview should work with all the variations / changes. Can you share more details on what exactly wasn't working when you changed it ?

3. Contact Section: That error happens when you are in preview mode and you haven't selected a contact form, if you select a contact form it will go away. We have already configured a contact form you just have to select it.

4. Have you checked our intro to Statamic video ? Give it a look I explain most of the stuff [here](https://www.youtube.com/watch?v=XBa6MWjCZCs)

Sorry but without giving me concrete errors I can't help you, also to answer for your question: Yes it is production ready, we have tested it out with multiple variations and one of them is running on our live site [here](https://www.landtamic.com)

Thanks,

**Comment by XmeGit** [comment]
Hello, 
thank you for your fast reply!

ad 4. I made all changes according to the video. 
ad 3. At Pages, I now played around without using the Preview (like demonstrated in the video/just made small changes Button size etc). I noticed no error so far. 
ad 2. True. Again, when I don't use Contact Form in Preview, there is no error.
ad 1. In Theme whatever I do, f.e. Menu Position - Center etc.. results to "The given data was invalid" in red. 

When are you planning an update for "Theme" Globals and Preview?

Thanks!

**Reply by lokmanm** [reply]
@XmeGit 

The error you are getting on Theme is a generic error but it should give you in red which field is invalid. You can also use the `Developer Console` and click on the error, it will give you the exact field ( if you cant see it visually ).

As for the contact form we will add a check if it doesn't exist to not bring up the error in live preview. 

As for an update on `Theme` Globals i need to pinpoint the exact error in order to see if its an error in our end or you are inputing data that is not supported.

Thanks

**Comment by XmeGit** [comment]
Hello!
"but it should give you in red which field is invalid." Please see the screenshots attached. When I deactivate the red fields, I still have an error.
[landtamic-error-screenshot.pdf](https://github.com/lucky-media/landtamic/files/8049295/landtamic-error-screenshot.pdf)

**Reply by lokmanm** [reply]
So the Image Link is throwing an error, you can delete that set if you are not using it.

If you are using it, requires an image to be present.

This is an issue with Statamic that will be resolved in version 3.3 and we will update it accordingly.

For now the best way is to delete the two ImageLink sets by going to the three dots and deleting it. 

[This is the error in question](https://github.com/statamic/cms/issues/5092)

**Comment by XmeGit** [comment]
Fine. I deleted the Image Links and the Globals/Theme didn't throw any errors anymore.
